### ISSUE TITLE
Cannot use 'in' operator to search...

### ISSUE THREAD

**Issue by julmot** [yarnpkg/yarn#1186]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

Doesn't add packages.

**If the current behavior is a bug, please provide the steps to reproduce.**

Run `$ yarn add bootstrap`

**What is the expected behavior?**

Should add Bootstrap.

**Please mention your node.js, yarn and operating system version.**
node.js: v5.6.0
npm: 3.10.9
yarn: 0.15.1
Windows: 7

**Log:**
```
yarn add v0.15.1
info No lockfile found.
[1/4] Resolving packages...
error TypeError: Cannot use 'in' operator to search for '~1.1.0' in undefined
    at C:\Users\julmot\AppData\Roaming\npm\node_modules\yarn\lib\resolvers\registries\npm-resolver.js:62:24
    at next (native)
    at step (C:\Users\julmot\AppData\Roaming\npm\node_modules\yarn\node_modules\babel-runtime\helpers\asyncToGenerator.js:17:30)
    at C:\Users\julmot\AppData\Roaming\npm\node_modules\yarn\node_modules\babel-runtime\helpers\asyncToGenerator.js:35:14
    at new Promise (C:\Users\julmot\AppData\Roaming\npm\node_modules\yarn\node_modules\core-js\library\modules\es6.promise.js:191:7)
    at C:\Users\julmot\AppData\Roaming\npm\node_modules\yarn\node_modules\babel-runtime\helpers\asyncToGenerator.js:14:12
    at Function.findVersionInRegistryResponse (C:\Users\julmot\AppData\Roaming\npm\node_modules\yarn\lib\resolvers\registries\npm-resolver.js:72:7)
    at C:\Users\julmot\AppData\Roaming\npm\node_modules\yarn\lib\resolvers\registries\npm-resolver.js:89:34
    at next (native)
    at step (C:\Users\julmot\AppData\Roaming\npm\node_modules\yarn\node_modules\babel-runtime\helpers\asyncToGenerator.js:17:30)
info Visit http://yarnpkg.com/en/docs/cli/add for documentation about this command.
```

**Comment by VsLoureiro** [comment]
I have the same problem when trying to install @types/xxx dependencies

**Comment by matrunchyk** [comment]
Same issue. Dump:
```
Arguments: 
  /usr/local/bin/node /Users/sergii/node_modules/bin/yarn install

PATH: 
  /usr/local/sbin:/Users/sergii/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/opt/ImageMagick/bin:/usr/local/MacGPG2/bin:/usr/local/rvm/bin:/Users/sergii/.composer/vendor/bin:/Users/sergii/node_modules/bin:/usr/local/bin:/usr/sbin:/sbin:/opt/local/bin:/opt/local/sbin

Yarn version: 
  0.16.1

Node version: 
  5.11.0

Platform: 
  darwin x64

npm manifest: 
  {
    "private": true,
    "scripts": {
      "prod": "gulp --production",
      "dev": "gulp watch"
    },
    "devDependencies": {
      "bootstrap-less": "^3.0.0",
      "gulp": "^3.9.1",
      "jquery": "^3.1.0",
      "laravel-elixir": "^5.0.0",
      "laravel-elixir-livereload": "^1.1.6"
    }
  }

yarn manifest: 
  No manifest

bower manifest: 
  No manifest

Lockfile: 
  No lockfile

Trace: 
  TypeError: Cannot use 'in' operator to search for '^1.1.0' in undefined
      at /Users/sergii/node_modules/lib/node_modules/yarn/lib/resolvers/registries/npm-resolver.js:62:24
      at next (native)
      at step (/Users/sergii/node_modules/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
      at /Users/sergii/node_modules/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:35:14
      at new Promise (/Users/sergii/node_modules/lib/node_modules/yarn/node_modules/core-js/library/modules/es6.promise.js:191:7)
      at /Users/sergii/node_modules/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:14:12
      at Function.findVersionInRegistryResponse (/Users/sergii/node_modules/lib/node_modules/yarn/lib/resolvers/registries/npm-resolver.js:72:7)
      at /Users/sergii/node_modules/lib/node_modules/yarn/lib/resolvers/registries/npm-resolver.js:89:34
      at next (native)
      at step (/Users/sergii/node_modules/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
```

**Comment by centsent** [comment]
Same issue. dump:
```
Arguments: 
  /Users/centsent/.nvm/versions/node/v6.8.0/bin/node /usr/local/bin/yarn add --dev mocha

PATH: 
  /Users/centsent/.rvm/gems/ruby-2.3.0/bin:/Users/centsent/.rvm/gems/ruby-2.3.0@global/bin:/Users/centsent/.rvm/rubies/ruby-2.3.0/bin:/Users/centsent/.nvm/versions/node/v6.8.0/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/centsent/.local/bin:/usr/local/opt/go/libexec/bin:/Users/centsent/workspace/vagrant/gopath/bin:/Users/centsent/.rvm/bin

Yarn version: 
  0.16.1

Node version: 
  6.8.0

Platform: 
  darwin x64

npm manifest: 
  {
    "name": "allus-api",
    "version": "4.0.0",
    "description": "RESTful api for https://www.wanmen.org",
    "main": "index.js",
    "scripts": {
      "start": "make",
      "test": "make test"
    },
    "repository": {
      "url": "git+https://github.com/wanmendaxue/allus-api.git",
      "type": "git"
    },
    "author": "centsent",
    "license": "ISC",
    "bugs": {
      "url": "https://github.com/wanmendaxue/allus-api/issues"
    },
    "homepage": "https://github.com/wanmendaxue/allus-api#readme",
    "devDependencies": {
      "babel-eslint": "^6.1.2",
      "eslint": "^3.3.1",
      "eslint-config-airbnb": "^10.0.1",
      "eslint-plugin-import": "^1.13.0",
      "eslint-plugin-jsx-a11y": "^2.1.0",
      "semver": "^5.3.0"
    },
    "dependencies": {
      "ramda": "^0.22.1"
    }
  }

yarn manifest: 
  No manifest

bower manifest: 
  No manifest

Lockfile: 
  No lockfile

Trace: 
  TypeError: Cannot use 'in' operator to search for '^5.4.0' in undefined
      at /usr/local/Cellar/yarn/0.16.1/libexec/lib/node_modules/yarn/lib/resolvers/registries/npm-resolver.js:62:24
      at next (native)
      at step (/usr/local/Cellar/yarn/0.16.1/libexec/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
      at /usr/local/Cellar/yarn/0.16.1/libexec/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:35:14
      at Promise.F (/usr/local/Cellar/yarn/0.16.1/libexec/lib/node_modules/yarn/node_modules/core-js/library/modules/_export.js:35:28)
      at /usr/local/Cellar/yarn/0.16.1/libexec/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:14:12
      at Function.findVersionInRegistryResponse (/usr/local/Cellar/yarn/0.16.1/libexec/lib/node_modules/yarn/lib/resolvers/registries/npm-resolver.js:72:7)
      at /usr/local/Cellar/yarn/0.16.1/libexec/lib/node_modules/yarn/lib/resolvers/registries/npm-resolver.js:89:34
      at next (native)
      at step (/usr/local/Cellar/yarn/0.16.1/libexec/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
```

**Comment by samccone** [comment]
fixed in #1356
